kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -192,13 +216,26 @@ private static Schema inferSchemaInternal(String name, InputStream incoming,
           throw new DatasetException("Found null value for required field: " +
               fieldName + " (" + types[i] + ")");
         }
-        fieldAssembler = fieldAssembler.name(fieldName)
-            .doc("Type inferred from '" + sample(values[i]) + "'")
-            .type(schema(types[i], false)).noDefault();
+        if (specifiedTypes != null && specifiedTypes.containsKey(fieldName)) {
```

Comment:
I removed a couple of comments that no longer apply.

I took a look at the original method and I don't think this should change the inference code. Instead, the assembly code here should use the given type to override `types[i]`. For each field, determine the type above (around where `foundNull` is set) and then pass the type into schema. This also removes the need to have so many blocks that check whether `specifiedTypes` has the field.

I would also like to make sure the specified type is compatible with the inferred type. If the inferred type is string but the specified type is int, then we know that's going to be a problem and should throw an exception.